llvm/lib/IR/IRBuilder.cpp

```diff
@@ -584,7 +584,8 @@ CallInst *IRBuilderBase::CreateMaskedLoad(Type *Ty, Value *Ptr, Align Alignment,
   if (!PassThru)
     PassThru = PoisonValue::get(Ty);
   Type *OverloadedTypes[] = { Ty, PtrTy };
-  Value *Ops[] = {Ptr, getInt32(Alignment.value()), Mask, PassThru};
+  unsigned Align32 = std::min<uint64_t>(Alignment.value(), 0x80000000ULL);
```

Comment:
I would rather use:
```
const Align LimitedAlign = std::min(Alignment, Align(32));
Value *Ops[] = {Ptr, getInt32(LimitedAlign.value()), Mask, PassThru};
```

and possibly encapsulate the logic into a function instead of copy/pasting the code.

Reply:
Sounds good. Is the 32 deliberate, or 1ULL<<31?

What about having a method in Align for getting the alignment limited to a 32bit value? Alignment.value32()? It would help encapsulate it.